src/vos/vos_common.c

```diff
@@ -705,19 +707,33 @@ vos_self_nvme_init(const char *vos_path, uint32_t tgt_id)
 	fd = open(nvme_conf, O_RDONLY, 0600);
 	if (fd < 0) {
 		rc = bio_nvme_init(NULL, VOS_NVME_NUMA_NODE, 0, 0,
-				   VOS_NVME_NR_TARGET, vos_db_get(), true);
+				   VOS_NVME_NR_TARGET, true);
 	} else {
 		rc = bio_nvme_init(nvme_conf, VOS_NVME_NUMA_NODE,
 				   VOS_NVME_MEM_SIZE, VOS_NVME_HUGEPAGE_SIZE,
-				   VOS_NVME_NR_TARGET, vos_db_get(), true);
+				   VOS_NVME_NR_TARGET, true);
 		close(fd);
 	}
 
 	if (rc)
 		goto out;
 
-	self_mode.self_nvme_init = true;
+	if (use_sys_db)
+		rc = vos_db_init(vos_path);
+	else
+		rc = vos_db_init_ex(vos_path, "self_db", true, true);
+	if (rc)
+		goto out;
```

Comment:
What's the purpose of moving these from vos_self_init() to here?

Reply:
will move back, this is history reason.